pkg/sync/sync_context.go

```diff
@@ -341,6 +331,19 @@ func (sc *syncContext) Sync() {
 		return
 	}
 
+	// delete all completed hooks which have appropriate delete policy
+	hooksPendingDeletion := tasks.Filter(func(task *syncTask) bool {
+		return task.isHook() && task.liveObj != nil && !task.running() && task.deleteOnPhaseCompletion()
+	})
+	for _, task := range hooksPendingDeletion {
+		if task.deleteOnPhaseCompletion() {
```

Comment:
Is this check necessary? It's already part of the task filter above.

Reply:
nice catch, thank you! the check is not required - removed.